File: reactive_mongodb_example_java/src/main/java/com/github/nkolytschew/data/repository/RatingDocumentRepository.java
package com.github.nkolytschew.data.repository;

import com.github.nkolytschew.data.document.RatingDocument;

import org.springframework.data.repository.query.Param;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;

import reactor.core.publisher.Flux;

/**
 * simple mongo repository.
 * <p>
 * provides the main CRUD operations.
 */
public interface RatingDocumentRepository extends ReactiveCrudRepository<RatingDocument, String> {
    /**
     * return all ratingsDocs with corresponding configurationId
     */
    Flux<RatingDocument> findAllByConfigurationIdOrderByCreationDateDesc(@Param("configurationId") String configurationId);
}

File: reactive_mongodb_example/src/main/kotlin/com/github/nkolytschew/data/repository/RatingDocumentRepository.kt
package com.github.nkolytschew.data.repository

import com.github.nkolytschew.data.document.RatingDocument
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.data.repository.query.Param
import reactor.core.publisher.Flux

/**
 * simple mongo repository.
 *
 * provides the main CRUD operations.
 */
interface RatingDocumentRepository : ReactiveMongoRepository<RatingDocument, String> {
  /**
   * return all ratingsDocs with corresponding configurationId
   */
  fun findAllByConfigurationIdOrderByCreationDateDesc(@Param("configurationId") configurationId: String): Flux<RatingDocument>
}